### PR TITLE
feat: implement editing and deleting messages & s-substitution

### PR DESCRIPTION
**Pull request by sewnie**
applies on-top of #313

### CODE REVIEW

**Comment by sewnie** [comment]
note: window is not re-drawn

File: cmd/run/message_input.go [hunk 1]
```diff
@@ -68,35 +68,64 @@ func (mi *MessageInput) sendAction() {
 		return
 	}
 
+	title := mi.GetTitle()
```
Comment (ayn2op) [inline cmd/run/message_input.go]: Move `title` to closest to where it is used.

File: cmd/run/message_input.go [hunk 2]
```diff
@@ -68,58 +68,75 @@ func (mi *MessageInput) sendAction() {
...
+		} else if strings.HasPrefix(title, "Editing") || substitution {
+			m, err := discordState.EditText(mainFlex.guildsTree.selectedChannelID, ms[mainFlex.messagesText.selectedMessage].ID, text)
+			if err != nil {
+				log.Println(err)
+				return
+			}
+
+			if err := discordState.MessageSet(m, true); err != nil {
+				log.Println(err)
+			}
```
Comment (ayn2op) [inline cmd/run/message_input.go]:
```suggestion
			err = discordState.MessageSet(m, true)
			if err != nil {
				log.Println(err)
				return
			}
```

File: ui/login_form.go [hunk 3]
```diff
@@ -13,12 +13,14 @@ import (
...
 	*tview.Form
 	Token chan string
+	Error chan error
 }
 
 func NewLoginForm() *LoginForm {
 	lf := &LoginForm{
 		Form:  tview.NewForm(),
 		Token: make(chan string, 1),
+		Error: make(chan error, 0),
```
Comment (ayn2op) [inline ui/login_form.go]: Channels are unbuffered by default. `0` is redundant here.
```suggestion
		Error: make(chan error),
```

File: ui/login_form.go [hunk 4]
```diff
@@ -13,12 +13,14 @@ import (
 type LoginForm struct {
 	*tview.Form
 	Token chan string
+	Error chan error
```
Comment (ayn2op) [inline ui/login_form.go:16]: This is not related to the PR, please remove it and open a new PR.
